atom/browser/api/atom_api_render_process_preferences.cc

```diff
@@ -19,18 +18,15 @@ namespace api {
 
 namespace {
 
-bool IsBrowserWindow(content::RenderProcessHost* process) {
+bool IsWebContents(v8::Isolate* isolate, content::RenderProcessHost* process) {
   content::WebContents* web_contents =
       static_cast<AtomBrowserClient*>(AtomBrowserClient::Get())->
           GetWebContentsFromProcessID(process->GetID());
   if (!web_contents)
     return false;
 
-  NativeWindow* window = NativeWindow::FromWebContents(web_contents);
-  if (!window)
-    return false;
-
-  return true;
+  auto api_web_contents = WebContents::CreateFrom(isolate, web_contents);
+  return !api_web_contents->IsRemote();
```

Comment:
This is also injecting content scripts to background pages, maybe we should add a new type of WebContents for background pages.

```javascript
const contents = webContents.create({type: 'backgroundPage'})
contents.getType === 'backgroundPage'
```